tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS ClassFeats (
+	class_id		UUID REFERENCES Classes,
+		NOT NULL
+	feat_id			UUID REFERENCES Feats
+		NOT NULL
+);
+
+--Features tables
+CREATE TABLE IF NOT EXISTS Features (
+	feature_id		UUID PRIMARY KEY,
+	description		TEXT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS ClassFeatures (
+	class_id		UUID REFERENCES Feats
+		NOT NULL,
+	feature_id		UUID REFERENCES Features
+		NOT NULL,
+	is_default		BOOLEAN
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS CharacterFeatures (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	feature_id		UUID REFERENCES Features
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS SubclassFeatures (
+	subclass_id		UUID REFERENCES Subclasses
+		NOT NULL,
+	feature_id		UUID REFERENCES Features
+		NOT NULL
+);
+
+--class proficiencies tables
+CREATE TABLE IF NOT EXISTS ClassWeaponsNotProficient (
+	class_id		UUID REFERENCES Classes
+		NOT NULL,
+	weapon_classes		WeaponClass[]
+);
+
+CREATE TABLE IF NOT EXISTS ClassArmorsNotProficient (
+	class_id		UUID REFERENCES Classes
+		NOT NULL,
+	armor_classes		ArmorClass[]
+);
+
+--Spells tables
+CREATE TABLE IF NOT EXISTS Spells (
+	spell_id		UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL,
+	level			INT
+		NOT NULL
+		CHECK (level >= 0),
+	school			MagicSchool
+		NOT NULL,
+
+	casting_time		INT
+		NOT NULL
+		CHECK (casting_time >= 0),
+	range			SpellRange
+		NOT NULL,
+	area			TEXT
+		NOT NULL,
+	duration_per_level	INT
+		NOT NULL
+		CHECK (duration_per_level > 0),
+	saving_throw		SaveThrow,
+	spell_resistance	BOOLEAN
+		NOT NULL,
+	description		TEXT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS CharacterSpells (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	spell_id		UUID REFERENCES Spells
+		NOT NULL,
+	casts			INT
+		NOT NULL
+		CHECK (casts >= 0)
+);
+
+CREATE TABLE IF NOT EXISTS SubclassSpells (
+	subclass_id		UUID REFERENCES Subclasses,
+	spell_id		UUID REFERENCES Spells,
+	casts			INT,
+	req_level		INT
+);
+
+CREATE TABLE IF NOT EXISTS DomainSpells (
+	domain_id		UUID REFERENCES Domains,
+	spell_id		UUID REFERENCES Spells,
+	casts			INT
+);
+
+CREATE TABLE IF NOT EXISTS SpellComponents (
+	spell_id		UUID REFERENCES Spells
+		NOT NULL,
+	item_id			UUID REFERENCES Items,
+
+	amount			INT,
+	component_type		ComponentType
+		NOT NULL
+);
+
+--Domain tables
+CREATE TABLE IF NOT EXISTS Domains (
+	domain_id		UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL,
+	description		TEXT
+		NOT NULL,
+	power_description	TEXT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS Subdomains (
+	domain_id		UUID REFERENCES Domains
+		NOT NULL,
+
+	name			TEXT
+		NOT NULL,
+	description		TEXT
+		NOT NULL
+);
+
+--Deity tables
+CREATE TABLE IF NOT EXISTS Deities (
+	deity_id		UUID PRIMARY KEY,
+	
+	name			TEXT
+		NOT NULL,
+	description		TEXT
+		NOT NULL,
+	favored_animals		TEXT
+);
+
+CREATE TABLE IF NOT EXISTS DeityDomains (
+	deity_id		UUID REFERENCES Deities
+		NOT NULL,
+	domain_id		UUID REFERENCES Domains
+		NOT NULL 	--no deities without domains
+);
```

Comment:
Is there a way to require that every deity has at least one domain? Some sort of `CHECK` or other constraint, perhaps?

Reply:
Not that I know of, sadly. I can look into this as an additional constraint to implement, but there doesn't seem to be a built-in way to ensure that every entry in Deities has an entry in the Domains table.

Reply:
Might be able to do something using triggers, but I'd have to look into that before I could say one way or another. Let's drop this for now and we can look at making it enforced later.